Aditi Kannan

January 04, 2021

VKC Lab

Dr. Kevin Bonham

<u>Summary of Progress and Future Steps</u>

The goal of this current project is to employ metagenomic analysis tools
to construct genomes for and further characterize *Bifidobacterium
infantis* (*B. infantis*)*,* a common colonizer of the human infant gut.
*B. infantis* is capable of digesting human milk oligosaccharides
(HMOs), which are the third most abundant solid component of human milk.
Interestingly enough, HMOs cannot be independently digested by infants,
and it seems as though HMOs are produced solely to sustain microbes such
as *B. infantis.* Additionally, the presence of *B. infantis* in the
infant gut has been positively associated with long-term human health.
It is our hope to use publicly available infant gut metagenomes in order
to assemble *B. infantis* genomes across populations around the world.
In doing this, we intend to analyze the genetic variation between
strains of the microbe, and further characterize the biogeographic
stratification of *B. infantis.*

The past summer and fall semester were used to lay a foundation for this
large project. Firstly, a list of papers with corresponding public
metagenomic databases was compiled and cataloged. Next, papers
discussing the more technical aspects of metagenomic assembly were read,
and a series of helpful tools were noted down. In order to assess the
quality of the public databases we collected, and to see if *B.
infantis* could be found, the Web Basic Local Alignment Search Tool
(WebBLAST) and the Joint Genome Institute’s Integrated Microbial Genomes
& Microbiomes (IMG/M) systems were trialed. Though some search results
were found using WebBLAST, it was determined that better and more
reliable results would be collected through a different method.

Due to this, I met with Lauren Tso ‘20 to learn more about the
computational tools she used to conduct metagenomic analysis for the
Environmental influences on Child Health Outcomes (ECHO) project. I
procured two files written in Python from her, and I am currently
working on debugging and using the file. Firstly, in order to be able to
work remotely through the lab’s computer, Hopper, I learned how to use
my computer’s Terminal to connect to Hopper. I then downloaded miniconda
for Python 3 onto Hopper, and proceeded to install all of the necessary
tools. I also made a git repository for this project, and have been
refamiliarizing myself with GitHub. Finally, I installed bowtie2 onto
Hopper, and hope to begin using this program as well.

Clearly, this project is still in its infant stages, and there is lots
of work to be done. My first order of business is to get Lauren’s files
running in order to compile a list of metagenomes containing *B.
infantis*. In order to familiarize myself with computational biology and
bioinformatics using Python, I am taking a Coursera class titled
“Biology Meets Programming: Bioinformatics for Beginners,” of which I
have completed 25%. Concurrently, I hope to research bowtie2, and begin
to use this program to see if it will be more useful for the purposes of
this project. If this is not sufficient, I will attempt to write julia
code to achieve the desired result. In addition, I plan on making a
figure to encapsulate this work.

After this, the list of metagenomic assembly tools previously compiled
will be consulted and reviewed in order to find a program or programs to
be used to construct the individual *B. infantis* genomes. This will
allow us to finally study the genetic variation between strains of *B.
infantis,* and to see if there are any connections between the genetic
variation and the location from which the strain was found. This has
potential to illustrate the differences that arise between the gut
microbiomes of infants from Westernized and non-Westernized lifestyles.
Additionally, since *B. infantis* is not found in all infant guts due to
many factors, this study can eventually allow us to observe whether the
presence of *B. infantis* could be connected to genetic or environmental
factors. The study of *Bifidobacterium infantis* through this project
will be exciting and thrilling, and can help us understand the
connection between this important infant gut microbe and long-term human
health.
